File: Backend/Domain/Model/UserModel.cs
using Domain.Entities;
using Domain.Enum;
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RoleType = Domain.Enum.RoleType;

namespace Domain.Model
{
    public class UserModel
    {
        public Guid? Id { get; set; }
        public string Email { get; set; } = default!;
        public string UserName { get; set; } = default!;
        public string? Password { get; set; } = default!;
        public RoleType? Role { get; set; } = default!;

        //Employer attributes
        public string? CompanyName { get; set; }

        //Student attributes
        public string? Name { get; set; }
        public string? University { get; set; }
        public Guid? StudyFieldId { get; set; }
        public DateTime? CreatedAt { get; set; } = DateTime.UtcNow;

        //Shared student and employer attributes
        public string? ProfileImagePath { get; set; }
        public Guid? LocationId { get; set; }
        public IFormFile? ProfileImage { get; set; }
    }
}

File: frontend/src/Interface/UserModel.ts
import { Role } from "../Enum/Role";

export interface UserModel {
  id: string | null | undefined;
  email: string | null;
  userName: string | null;
  password: string | null;
  role: Role;

  // Fields for employer
  companyName:string|null;
  //Fields for student
  name:string|null;
  university:string|null;
  studyFieldId: string|null;
  createdAt:string|null;
  //Fields for student and employer
  profileImagePath: string|null;
  locationId:string|null;
  profileImage?: File;
}